devicemodel/hw/vga.c

```diff
@@ -325,28 +324,46 @@ vga_mem_rd_handler(struct vmctx *ctx, uint64_t addr, void *arg1)
 		/*
 		 * extended mode: base 0xa0000 size 128k
 		 */
-		offset -=0xa0000;
-		offset &= (128 * KB - 1);
-		break;
+		if (offset >= 0xa0000) {
+			offset -=0xa0000;
+			offset &= (128 * KB - 1);
+			break;
+		} else {
+			return 0;
```

Comment:
We need to print some logs. BTW, for the invalid mmio, we usually return 0xFFFFFFFF.